src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -291,14 +343,23 @@ export class DashboardState {
     return this.appState.panels;
   }
 
+  updatePanel(panelIndex, panelAttributes) {
+    const updatedPanel = this.getPanels().find((panel) => panel.panelIndex === panelIndex);
+    Object.assign(updatedPanel, panelAttributes);
```

Comment:
Kim recently merged a PR moving away from using `Object.assign` to using the `...` syntax. Do we need to be mutating the updatedPanel object directly and continue using `Object.assign` or can we switch this over?